### ISSUE TITLE
ER: GHA to create issues to manage CodeQL alerts

### ISSUE THREAD

**Issue by roslynwythe** [hackforla/website#5007]
### Emergent Requirement - Problem
We require issues to track each CodeQL alert.  The issue that tracks a CodeQL alert will require the analysis of the alert and either a code change to "fix" the alert or a recommendation to "dismiss" the alert (for example, as a false positive).  Currently, the tracking issues have been created manually, but it would be desirable to generate the issues automatically. 

### Details

- If any GitHub issue contains a CodeQL alert URL in a task item, the issue will become a tracking issue to the alert.  
- In this ER we are only concerned with alerts raised on production code, not on alerts within pull requests.  
- Since GitHub will block merging of any PR with a high or critical severity, the only new alerts will result from medium or lower severity alerts, or from changes in queries.  

### Issue you discovered this emergent requirement in
- #4886

### Date discovered 
6/23/2023

### Did you have to do something temporarily
- [ ] YES
- [x] NO

### Who was involved
@SAUMILDHANKAR @t-will-gillis @roslynwythe 

### What happens if this is not addressed
Tracking issues will have to be created manually for each CodeQL alerts, and new alerts may go unnoticed.  

### Resources
- [GitHub CodeQL documentation](https://codeql.github.com/docs/codeql-overview/about-codeql/)
-  #5060

### Recommended Action Items
- [ ] Make a new issue
- [ ] Discuss with team
- [ ] Let a Team Lead know

### Potential solutions [draft]

Develop a GHA that examines every open CodeQL alert for each one, check for a  tracking issue and for the source.   If a CodeQL alert from `gh-pages` does not have a tracking issue, create a new issue based on the template below, move the issue to "New Issue Approval" column with "ready for dev lead" label.

**Comment by roslynwythe** [comment]
### Template for new issues for tracking CodeQL alerts:  (Title:  Resolve CodeQL Alert INSERT-ALERT-ID)

### Prerequisite
1. Be a member of Hack for LA. (There are no fees to join.) If you have not joined yet, please follow the steps on our [Getting Started page](https://www.hackforla.org/getting-started).
2. Before you claim or start working on an issue, please make sure you have read our [How to Contribute to Hack for LA Guide](https://github.com/hackforla/website/blob/7f0c132c96f71230b8935759e1f8711ccb340c0f/CONTRIBUTING.md).

### Overview
As developers. we need to analyze [CodeQL query alert INSERT-ALERT-ID](https://github.com/hackforla/website/security/code-scanning/INSERT-ALERT-ID) and to either recommend dismissal of the alert or update the code to resolve the alert.    

### Action Items
- [ ] DO NOT DISMISS ANY ALERTS.  Dismissal of alerts should be done by dev leads only after review of the recommendation
- [ ] Browse to link below and read the contents.  Click "See More" to view Recommendations, Examples and References.  
   - [ ] https://github.com/hackforla/website/security/code-scanning/INSERT-ALERT-ID 
- [ ] In a comment in this issue, add your analysis and recommendations with details.  The recommendation can be one of the following: `dismiss as test`, `dismiss as false positive`, `dismiss as won't fix`, or `update code`.  An example of a 'false positive' is a report of a JavaScript syntax error that is caused by markdown or liquid symbols such as `---` or `{%`.  
- [ ] If the recommendation is to update code:
   - [ ] create an issue branch and proceed with the code update
   - [ ] Use docker to test locally, ensuring that there are no changes to any affected webpage(s)
   - [ ] proceed with pull request in the usual manner 
- [ ] If the recommendation is to dismiss, describe your reason for dismissal in the comment, then move the issue to `Questions/In Review` and apply the label `ready for dev lead`.  


### For merge team/dev lead
- [ ] If recommendation to dismiss is approved, dismiss the alert with a comment, then close the issue as completed.
  
### Resources/Instructions
- [GitHub CodeQL documentation](https://codeql.github.com/docs/codeql-overview/about-codeql/)

**Comment by ExperimentsInHonesty** [comment]
I am signing off on the template above.

**Comment by roslynwythe** [comment]
- Created #5059
